iron-list.html

```diff
@@ -1390,11 +1390,14 @@
 
       forceUpdate = forceUpdate || this._scrollHeight === 0;
       forceUpdate = forceUpdate || this._scrollPosition >= this._estScrollHeight - this._physicalSize;
+      // Probably a bug that style.height is a string like '100px' and it is being compared to
+      // a number?
       forceUpdate = forceUpdate || this.grid && this.$.items.style.height < this._estScrollHeight;
 
       // Amortize height adjustment, so it won't trigger large repaints too often.
       if (forceUpdate || Math.abs(this._estScrollHeight - this._scrollHeight) >= this._optPhysicalSize) {
         this.$.items.style.height = this._estScrollHeight + 'px';
+        this.updateViewportBoundaries();
```

Comment:
@danbeam Looking through this example, I don't see anything that would let the list (at least cr-network-list) know that is had come into view. Maybe there's something at the page level, further up than I chose to investigate, but if you're really placing the list on the DOM while invisible and then making it visible, you have to wait till it is visible for `this.notifyResize()` to be able to actually do its job. In Crome Settings' case, this might be a perfect opportunity to employ an Intersection Observer, though that may just be me wanting a good reason to use one. I do see how the one forced recalc in this PR is better than the approach in the issue you linked, being it would be doing a lot more mathing of the page (seemingly over and over again in empty lists) but as they say "a stitch in time saves nine".